### PR TITLE
fix: allow task items to be parsed when only having `<li data-checked` instead of only when `<li data-checked="true"` (re-fix of #5366)

### PR DESCRIPTION
**Pull request by baseballyama**
## Changes Overview

When I modified the code in response to [the comments](https://github.com/ueberdosis/tiptap/pull/5366#discussion_r1682594145) in PR #5366, somehow I made the wrong modification. I am very sorry.🙏🙏🙏 @nperez0111 

This PR will be correctly amended according to this comment.
I have confirmed that it works correctly with both the REPL created in #5366 and my own product code.

I will take care to avoid similar carelessness.🙏

## Testing Done
I tested it by manual. 

## Checklist
- [x] I have created a [changeset](https://github.com/changesets/changesets) for this PR if necessary.
- [x] My changes do not break the library.
- [ ] I have added tests where applicable.
- [x] I have followed the project guidelines.
- [x] I have fixed any lint issues.

## Related Issues
https://github.com/ueberdosis/tiptap/pull/5366

### CODE REVIEW

**Comment by baseballyama** [comment]
@bdbch 

Did you force pushed some commit to https://github.com/baseballyama/tiptap/commits/develop/ ? (This is my forked repo)
image

Maybe did you intend to push it to [ueberdosis:develop](https://github.com/ueberdosis/tiptap/tree/develop)?

File: packages/extension-task-item/src/task-item.ts [hunk 1]
```diff
@@ -68,7 +68,7 @@ export const TaskItem = Node.create<TaskItemOptions>({
         parseHTML: element => {
           const dataChecked = element.getAttribute('data-checked')
 
-          return dataChecked == null || dataChecked === 'true'
```
Comment (bdbch) [inline packages/extension-task-item/src/task-item.ts:71]: Should we still keep the null check here?

```ts
return dataChecked === null || dataChecked === '' || dataChecked === 'true'
```

**Review by bdbch — COMMENTED** [review]
LGTM - just not sure if we need the null check or not.

**Comment by bdbch** [comment]
@baseballyama I force pushed on our `develop` to move the commit you mentioned over to `next`.

You probably have to rebase your branch to the current develop here.

File: packages/extension-task-item/src/task-item.ts (hunk 1 above)
Comment (baseballyama) [reply packages/extension-task-item/src/task-item.ts:71]: No I believe we shouldn't.

Case1: `<li>`, `dataChecked` will be `null`,
Case2: `<li data-checked>`, `dataChecked` will be `''`,
Case3: `<li data-checked="true">`, `dataChecked` will be `'true'`,
Case3: `<li data-checked="false">`, `dataChecked` will be `'false'`,

Case1 and case4, `checked` attribute should be recognize as `false`,
Case 2 and case3, `checked` attribute should be recognize as `true`,

**Comment by baseballyama** [comment]
> I force pushed on our develop to move the commit you mentioned over to next.
> You probably have to rebase your branch to the current develop here.

Understood.
I think current diff is looks fine.

https://github.com/ueberdosis/tiptap/compare/develop...baseballyama:fix/5366

**Review by bdbch — APPROVED** [review]

**Pull request merged**